uint8 txdata
uint8 address